src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -281,8 +281,13 @@ public void onSubscribe(Flow.Subscription subscription) {
 
         @Override
         public void onNext(List<ByteBuffer> items) {
+            int size = Utils.remaining(items, Integer.MAX_VALUE);
             try {
-                out.write(items.toArray(Utils.EMPTY_BB_ARRAY));
+                long bytesWritten = 0;
+                ByteBuffer[] buffers = items.toArray(Utils.EMPTY_BB_ARRAY);
+                while (bytesWritten < size) {
+                    bytesWritten += out.write(buffers);
```

Comment:
Should we handle the potential case of long value overflow, after addition here, which could then cause the while loop to misbehave?

Reply:
I replaced usages with Math::addExact, and generalised the catch to handle arithmetic exceptions too

Reply:
That's the reason I originally didn't comment on the size limitation to Integer.MAX_VALUE - this is an acceptable limitation and it avoids to have to care about overflows here if size <=  Integer.MAX_VALUE. Integer.MAX_VALUE shoudln't be reached in the context where the publisher is the HttpClient.

If we want to allow longs however, we should really use Math.addExact all the way down - including in Utils::remaining.

Reply:
I don't think we can ever overflow long here; ByteBuffer.remaining can not exceed Integer.MAX_VALUE, and items.toArray cannot create an array with more elements than integer.MAX_VALUE.

Reply:
Ah! Good point...

Reply:
True, however Math.addExact is fine here.